src/initial_conditions/initial_conditions.jl

```diff
@@ -351,10 +376,80 @@ function (initial_condition::RisingThermalBubbleProfile)(params)
     return local_state
 end
 
+overwrite_initial_conditions!(initial_condition::InitialCondition, args...) =
+    (return nothing)
+
+"""
+    overwrite_initial_conditions!(initial_condition, Y, thermo_params, config)
+Given a prognostic state `Y`, an `initial condition` (specifically, where
+initial values are assigned from interpolations of existing datasets), a `thermo_state`,
+and the `config::AtmosConfig` object, this function overwrites the default initial condition
+and populates prognostic variables with interpolated values using the `SpaceVaryingInputs`
+tool. 
+"""
+function overwrite_initial_conditions!(
```

Comment:
Could you make this function independent of the `parsed_args`?

Reply:
In the current implementation there is a check against the `moisture` model type, which assigns additional prognostic vars - right now it seems the easiest (and perhaps only way) to apply this check is through the `config` object : If there is a cleaner way to do this I'm happy to change this.

Reply:
Now uses `hasproperty()` on `Y.c` ; removes `config` as an arg.